src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-update.component.ts

```diff
@@ -73,29 +78,65 @@ export class FileUploadExerciseUpdateComponent implements OnInit {
         this.activatedRoute.data.subscribe(({ fileUploadExercise }) => {
             this.fileUploadExercise = fileUploadExercise;
             this.backupExercise = cloneDeep(this.fileUploadExercise);
-            this.isExamMode = this.fileUploadExercise.exerciseGroup !== undefined;
-            if (!this.isExamMode) {
-                this.exerciseCategories = this.fileUploadExercise.categories || [];
-                this.courseService.findAllCategoriesOfCourse(this.fileUploadExercise.course!.id!).subscribe({
-                    next: (categoryRes: HttpResponse<string[]>) => {
-                        this.existingCategories = this.exerciseService.convertExerciseCategoriesAsStringFromServer(categoryRes.body!);
-                    },
-                    error: (error: HttpErrorResponse) => onError(this.alertService, error),
-                });
-            }
-            // Exam exercises cannot be not included into the total score
-            if (this.isExamMode && this.fileUploadExercise.includedInOverallScore === IncludedInOverallScore.NOT_INCLUDED) {
-                this.fileUploadExercise.includedInOverallScore = IncludedInOverallScore.INCLUDED_COMPLETELY;
-            }
-
-            this.saveCommand = new SaveExerciseCommand(this.modalService, this.popupService, this.fileUploadExerciseService, this.backupExercise, this.editType, this.alertService);
+            getCourseId(fileUploadExercise);
+            this.examCourseId = getCourseId(fileUploadExercise);
         });
 
         this.activatedRoute.queryParams.subscribe((params) => {
             if (params.shouldHaveBackButtonToWizard) {
                 this.goBackAfterSaving = true;
             }
         });
+        this.activatedRoute.url
```

Comment:
Can you use some helper methods to keep this section shorter and less indented?

Reply:
I extracted these long if else parts into seperate methods.